File: Reto # - 1/C++/Reto1.cpp

//DeathbatO

#include <iostream>

using namespace std;

int main(){
    // Declaracion de variables que se usaran en el codigo
    int a, b, c, d, t, x, k, num, divi;
    bool bite1, bite2;
    char vocal;

    ////-------- Operadores ---------////

    //-- Aritmeticos --//
    a = 5;
    b = 10;
    short sum = a + b;
    short res = a - b;
    short mult = a * b;
    short div = a / b;
    short mod = a % b;
    /* Con el siguiente print se podra visualizar el resultado de cualquier
    operacion realizada en esta seccion reemplazando la que esta como ejemplo*/
    cout<<"Ejemplo de Operador Aritmetico (Modulo): "<<mod<<endl;


    //-- Comparativos --//
    c = 6;
    d = 7;
    bool OperadorIgual = (c == d);
    bool OperadorDesigual = (c != d);
    bool OperadorMayor = (c > d);
    bool OperadorMenor = (c < d);
    bool OperadorMayoroIgual = (c >= d);
    bool OperadorMenoroIgual = (c <= d);
    /* Con el siguiente print se podra visualizar el resultado de cualquier
    operacion realizada en esta seccion reemplazando la que esta como ejemplo*/
    cout<<"Ejemplo de operador comparativo (Menor): "<<OperadorMenor<<endl;//True (1)


    //-- Logicos --//
    bite1 = true;
    bite2 = false;
    bool OperadorAnd = (bite1 && bite2);
    bool OperadorOr = (bite1 || bite2);
    bool OperadorNot = (!bite1);
    /* Con el siguiente print se podra visualizar el resultado de cualquier
    operacion realizada en esta seccion reemplazando la que esta como ejemplo*/
    cout<<"Ejemplo de operador logico (And): "<<OperadorAnd<<endl;//False (0)


    //-- Bits --//
    short OpBitsRS = 10 << 2; // Rigth Shift (Desplazamiento a la derecha)
    short OpBitsLS = 10 >> 2; // Left Shift (Desplazamiento a la izquierda)
    short OpBitsXor = 10 ^ 2; // Operador XOR 4 ^ 2 = 6
    /* Con el siguiente print se podra visualizar el resultado de cualquier
    operacion realizada en esta seccion reemplazando la que esta como ejemplo*/
    cout<<"Ejemplo de operador de Bits (OpBitsXor): "<<OpBitsXor<<endl;


    ////-------- Estructuras de control ---------////

    //-- Condicionales --//
    cout<<"Por favor digita un numero entero para compararlo con 20: ";
    cin>>t;
    if(t > 20){
        cout<<"El numero es mayor a 20"<<endl;
    }else if(t < 20){
        cout<<"El numero es menor a 20"<<endl;
    }else{
        cout<<"El numero es 20"<<endl;
    }


    cout<<"Por favor ingresa una vocal en mayuscula: ";
    cin>>vocal;

    switch (vocal){
    case 'A':
        cout<<"Vocal A"<<endl;
        break;
    case 'E':
        cout<<"Vocal E"<<endl;
        break;
    case 'I':
        cout<<"Vocal I"<<endl;
        break;
    case 'O':
        cout<<"Vocal O"<<endl;
        break;
    case 'U':
        cout<<"Vocal U"<<endl;
        break;
    
    default:
        cout<<"Eso no era una vocal o no estaba en mayuscula: "<<endl;
        break;
    }


    //-- Iterativas --//
    //Imprime los numero pares desde el 0 hasta el 12
    for(int i = 0; i <= 12; i++){
        if(i % 2 == 0){
            cout<<i<<" ";
        }
    }
    cout<<""<<endl;

    //Imprime los numeros multiplos de 3 desde el 0 hasta el 15
    k = 0;
    while(k <= 15){
        if(k % 3 == 0){
            cout<<k<<" ";
        }
        k++;
    }
    cout<<""<<endl;


    //-- Excepciones --//
    cout<<"Ingresa un numero para que 10 sea dividido por el: "<<endl;
    cin>>num;
    try{
        if(num == 0){
            throw std::runtime_error("No podemos dividir entre 0");
        }
        divi = 10 / num;
        cout<<"El resultado de la divicion es: "<<divi<<endl;
    }catch(const std::exception& e){
        std::cerr<<"Error: "<<e.what()<<"\n"<<endl;
    }


    ////-------- Reto Extra ---------////

    //-- Ejercicio con For --//

    cout<<"Ejercicio con For: "<<endl;
    for(int i = 10; i < 56; i++){
        if(i % 2 == 0 && i != 16 && i % 3 == 0){
            cout<<i<<" ";
        }
    }
    cout<<""<<endl;

    //-- Ejercicio con while --//
    cout<<"Ejercicio con While: "<<endl;
    k = 10;
    while(k < 56){
        if(k % 2 == 0 && k != 16 && k % 3 == 0){
            cout<<k<<" ";
        }
        k++;
    }

}

File: Reto # - 1/Python/Reto1.py

# DeathbatO

### Operadores ###

# Aritmeticos
a = 5
b = 10
suma = a + b
resta = a - b
mult = a * b
division = a / b
divEntera = a // b
modulo = a % b
potencia = a ** b
# El siguiente print muestra en consola el resultado de la resta como ejemplo
print(f"Ejemplo de operador aritmetico (operacion entre {a} y {b}): {resta}")


## Comparacion
c = 9
d = 11
igualdad = (c == d)
desigualdad = (c != d)
mayorque = (c > d)
menorque = (c < d)
mayoroigual = (c >= d)
menoroigual = (c <= d)
# El siguiente print muestra por consola el resultado de desigualdad como ejemplo
print(f"Ejemplo de Comparacion (Desigualdad): {desigualdad}") #True


## Operadores Logicos
bite1 = True
bite2 = False
operadorAnd = bite1 and bite2 #False
operadorOr = bite1 or bite2 #True
operadorNot = not bite1 #False
# El siguiente print muestra por consola el resultado de la operacion And
print(f"Ejemplo de operadores logicos (And): {operadorAnd}")


## Operadores de asignacion
#Asignacion simple
t = 100 # se asigna un valor de 100 a la variable 't'
#Asignacion de suma
t += 2 # Toma el valor que ya tenia y le suma 2
#Asignacion de resta
t -= 2 # Toma el valor que ya tenia y le resta 2
#Asignacion de multiplicacion
t *= 2 # Toma el valor que ya tenia y le multiplica 2
#Asignacion de division
t /= 2 # Toma el valor que ya tiene y lo divide entre 2
#Asignacion de divicion entera
t //= 2 # Toma el valor que ya tiene y hace la division entera entre 2
#Asignacion de potencia
t **= 2 #Toma el valor que ya tiene y hace la potencia al cuadrado


## Operadores de identidad
obj1 = 2
obj2 = 1

operdorIs = obj1 is obj1 # True por que son el mismo objeto
operdorIs2 = obj1 is obj2 # False por que no son el mismo objeto
operdaorIsnot = obj1 is not obj1 # False por que son el mismo objeto
operadorIsNot2 = obj1 is not obj2 # True por que no son el mismo objeto
# Este print muestra por consola el ejemplo del operadorIsNot2 (True)
print(f"Ejemplo de operador de identidad (Is Not): {operadorIsNot2}")


## Operadores de pertenencia
lista = [10,50,40,20,30]
operdorIn = (15 in lista) #False por que el elemento 15 no esta en la lista
opeadorNotIn = (25 not in lista) #True por que el elemento 25 no esta en la lista
print(f"Ejemplo de operdaor de pertenencia (Operdaor In): {operdorIn}")


## Operdaores de Bits
a = 1
b = 1

OpBitAnd = a & b
OpBitOr = a | b
OpBitNot = ~a
OpBitXor = a ^ b
OpBitLS = a << 2 # Left Shift (Desplazamiento a la izquierda)
OpBitsRS = b >> 2 # Rigth Shift (Desplazamiento a la derecha)
print(f"Ejemplo de operador de Bits (Or): {OpBitOr}")


### Estructuras de Control ###

## Condicionales
num = int(input("Ingrese un numero para compararlo con el numero 20: "))

if num > 20:
    print("El numero es mayor a 20")
elif num < 20:
    print("El numero es menor a 20")
else:
    print("El numero es 20")

## Iterativas
    
for i in range(10):
    print(i)

j = 5; 
while j > 0:
    print(j)
    j-=1

## Excepciones
    
while True:
    # try, except, finally
    try:
        x = int(input("Por favor ingrese un numero: "))
        break
    except ValueError:
        print("No ingresaste un numero, por favor digitalo bien")
    finally:
        print("Ahora si, continuamos")

    # Raise    
    x = "Entrada de texto"
    if not type(x) is int:
        raise TypeError("Solo se permiten numeros enteros")


### Reto Extra ###

## Ejercicio con for
    
for e in range(10, 56):
    if e % 2 == 0 and e != 16 and e % 3 ==0:
        print(e)

## Ejercicio con while
k = 10
while k <= 55:
    if k % 2 == 0 and k != 16 and k % 3 ==0:
        print(k)
    k += 1